### ISSUE TITLE
Standardize error messages

### ISSUE THREAD

**Issue by spawnia** [nuwave/lighthouse#655]
Right now the error messages that are thrown are pretty messy.

We should standardize them to make them more predictable and uniform.

- Should we terminate messages with a dot `.`?
- How do we quote things like class names? Backticks, Single or double quotes, square brackets, not at all?

**Comment by olivernybroe** [comment]
Thumbs up on this comment if you think we should terminate with dot.

**Comment by alberthaff** [comment]
@olivernybroe I agree. This definitely makes longer errors messages easier to read.

I think it would also make errors easier to read, if we placed URLs referring to docs, etc. at the very end.
```

// Current Exception message

Error
'Could not find a valid map, be sure to conform to GraphQL multipart request specification: https://github.com/jaydenseric/graphql-multipart-request-spec'


// URL to docs at the end

Error
'Could not find a valid map. Be sure to conform to GraphQL multipart request specification. Instead got: null. (see: https://github.com/jaydenseric/graphql-multipart-request-spec).'
```

**Comment by JasonTheAdams** [comment]
Hi! Jumping in here to see what I can find. 😃

I'm starting with an audit of how exceptions are currently generated, as well as the messages they provide.

### Test-Only Exceptions
The following are exceptions that are only thrown during tests. I suggest creating a `tests/Exceptions` directory for these. Exceptions within `src/Exceptions` should be limited to exceptions used during runtime.

- [AuthenticationException](https://github.com/nuwave/lighthouse/blob/master/src/Exceptions/AuthenticationException.php)
- 

### Unused Error Code
The following are classes/exceptions that do not currently have any use within the codebase.

- [HasErrorBuffer (trait)](https://github.com/nuwave/lighthouse/blob/master/src/Support/Traits/HasErrorBuffer.php)
- [HasErrorBuffer (interface)](https://github.com/nuwave/lighthouse/blob/master/src/Support/Contracts/HasErrorBuffer.php)
- [ErrorBuffer](https://github.com/nuwave/lighthouse/blob/master/src/Execution/ErrorBuffer.php)

Note on `ErrorBuffer`: It is used by the trait, which is not used. It is used by the interface, which is checked for in `FieldFactory` but is never implemented.

---

This is a work in progress. I'm still doing research and will edit this comment over the next couple days with more details.

**Comment by spawnia** [comment]
@JasonTheAdams looks great so far!

Another thing i would like to change is to get rid of the `DirectiveException` in favour of `DefinitionException` - the latter should be used to clearly indicate when the error originates from a faulty definition.

**Comment by algeriarocker** [comment]
the right way to fetch lighthous api with multiple formdata
`    var myHeaders = new Headers();
    myHeaders.append("Authorization", "Bearer " + localStorage.getItem("cnmtoken"));
    
    var formdata = new FormData();
    formdata.append("operations", "{\n        \"query\": \"mutation Upload($image: Upload!, $file: Upload!,$id:String!) { uploadFiles(image: $image, file: $file, id:$id) }\",\n        \"variables\": {\n            \"image\": null,\n            \"file\": null,\n\"id\":\""+postid+"\"\n        }\n    }");
    formdata.append("map", "{\n        \"0\": [\"variables.image\"],\n        \"1\": [\"variables.file\"]\n    }");
    formdata.append("0", cover);
    formdata.append("1", file);
    
    var requestOptions = {
      method: 'POST',
      headers: myHeaders,
      body: formdata,
      redirect: 'follow'
    };
    
    fetch(API, requestOptions)
      .then(response => response.text())
      .then(result => console.log(result))
      .catch(error => console.log('error', error));


`
